Umee-Bots/lendingPool-reentrancy/src/agent.ts

```diff
@@ -0,0 +1,38 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+
+import utils from "./utils";
+
+export const handleTransaction =
+  (contractAddress: string, reenterancyFunctionsSignatures: string[]): HandleTransaction =>
+  async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const { traces } = txEvent;
+    for (let i = 0; i < traces.length; i++) {
+      const trace = traces[i];
+      const depth = trace.traceAddress.length;
+
+      if (trace.action.to === contractAddress) {
+        let j;
+
+        for (j = i + 1; j < traces.length; j++) {
+          if (traces[j].traceAddress.length === depth) {
+            i = j;
```

Comment:
Also, this should be `i = j - 1` right? Otherwise the element at index `j` will be ignored.